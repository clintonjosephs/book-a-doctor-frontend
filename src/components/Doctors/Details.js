import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useParams, Link } from 'react-router-dom';
import { fetchOneDoctor } from '../../redux/bookDoctor/doctorThunks';
import classes from './Detail.module.css';

function Details() {
  const { id } = useParams();
  const dispatch = useDispatch();
  const { doctor } = useSelector((state) => state.bookDoctorReducer);

  useEffect(() => {
    dispatch(fetchOneDoctor(id));
  }, [dispatch]);

  console.log(doctor);

  return (
    <section className={classes.doctor}>
      <div className={classes.detail}>
        <div className={classes.photo}>
          <img alt="doctor" src={doctor.image_url} />
        </div>
        <div className={classes.info}>
          <h2 className={classes.name}>
            { doctor.name }
          </h2>
          <div className={classes.description}>
            { doctor.description }
          </div>
          <div className={`${classes.item} ${classes.neutral}`}>
            <div>City</div>
            <div className={classes.city}>
              { doctor.city }
            </div>
          </div>
          <div className={`${classes.item}`}>
            <div>Specialization</div>
            <div className={classes.specialization}>
              { doctor.specialization }
            </div>
          </div>
          <div className={`${classes.item} ${classes.neutral}`}>
            <div>Cost per day</div>
            <div className={classes.cost_per_day}>
              $
              { doctor.cost_per_day }
            </div>
          </div>
          <div className={classes.reserve}>
            <button type="button" className={classes.btn}>
              <i className="fa-solid fa-gear" />
              Reserve
              <i className="fa-regular fa-circle-right" />
            </button>
          </div>
        </div>
      </div>
      <div className={classes.back}>
        <Link to="/">
          <button type="button" className={classes.btn1}>
            <i className="fa-solid fa-caret-left" />
          </button>
        </Link>
      </div>

      <div />

    </section>
  );
}

export default Details;
