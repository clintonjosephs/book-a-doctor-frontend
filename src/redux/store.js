import { applyMiddleware, combineReducers, createStore } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import bookDoctorReducer from './bookDoctor/doctorActions';
import userReducer from './user/userActions';
import reservationReducers from './reservation/reservationActions';

const reducer = combineReducers({
  bookDoctorReducer,
  userReducer,
  reservationReducers,
});

// creating a store
const store = createStore(reducer, applyMiddleware(logger, thunk));

export default store;
