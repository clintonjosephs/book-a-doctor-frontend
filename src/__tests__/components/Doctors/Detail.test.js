import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import store from '../../../redux/store';
import Details from '../../../components/Doctors/Details';

describe('Countries snapshot', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <Details />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
