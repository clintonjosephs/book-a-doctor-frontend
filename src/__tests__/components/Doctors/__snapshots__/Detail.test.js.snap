// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Countries snapshot renders correctly 1`] = `
<section
  className="doctor"
>
  <div
    className="detail"
  >
    <div
      className="photo"
    >
      <img
        alt="doctor"
      />
    </div>
    <div
      className="info"
    >
      <h2
        className="name"
      />
      <div
        className="description"
      />
      <div
        className="item neutral"
      >
        <div>
          City
        </div>
        <div
          className="city"
        />
      </div>
      <div
        className="item"
      >
        <div>
          Specialization
        </div>
        <div
          className="specialization"
        />
      </div>
      <div
        className="item neutral"
      >
        <div>
          Cost per day
        </div>
        <div
          className="cost_per_day"
        >
          $
        </div>
      </div>
      <div
        className="reserve"
      >
        <button
          className="btn"
          type="button"
        >
          <i
            className="fa-solid fa-gear"
          />
          Reserve
          <i
            className="fa-regular fa-circle-right"
          />
        </button>
      </div>
    </div>
  </div>
  <div
    className="back"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <button
        className="btn1"
        type="button"
      >
        <i
          className="fa-solid fa-caret-left"
        />
      </button>
    </a>
  </div>
  <div />
</section>
`;
